twoliter/src/tools.rs

```diff
@@ -0,0 +1,138 @@
+use crate::common::fs;
+use crate::tools_hash::TOOLS_HASH;
+use anyhow::{Context, Result};
+use flate2::read::GzDecoder;
+use log::{debug, trace, warn};
+use std::path::Path;
+use tar::Archive;
+
+const TARBALL_DATA: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/tools.tar.gz"));
```

Comment:
If there's a decent crate for zstd, that would be better than gzip in terms of compressed size and decompression speed.